spec/lib/query/query_payments_spec.rb

```diff
@@ -822,6 +822,31 @@ def generate_donation(h)
         expect(result[:data].count).to eq 4
         expect(result[:data]).to_not satisfy {|i| i.any?{|j| j['id'] == donation_result_tomorrow['payment']['id']}}
       end
+
+      context 'when filtering by campaign' do 
+        it 'returns 2 campaign results' do 
+          donation_result_today
+          donation_result_yesterday
+          donation_result_tomorrow
+
+          result = QueryPayments::full_search(nonprofit.id, {campaign_id: campaign.id})
+          expect(result[:data].count).to eq 2
+          expect(result[:data]).to_not satisfy {|i| i.any?{|j| j['id'] == donation_result_tomorrow['campaign']['id']}}
+        end 
+      end 
+
+      context 'and sorts results' do 
+        it 'returns sorted campaign results' do 
+          donation_result_today
+          donation_result_tomorrow
+          donation_result_yesterday
+
+          result = QueryPayments::full_search(nonprofit.id, {campaign_id: campaign.id}).sort 
```

Comment:
@camckin10 so the issue isn't that we can't sort the result after it's been completed. The issue is whether the full_search method itself will perform the sort. It does that on the database. The method you're using gets the result from the database and then performs a sort. That's not really testing what we need.

Reply:
Ohhh I see! Thank you. I will update the spec.